lib/Onyx.d.ts

```diff
@@ -0,0 +1,278 @@
+import {Component} from 'react';
+import {PartialDeep} from 'type-fest';
+import * as Logger from './Logger';
+import {DeepRecord, Key, CollectionKey, Selector, Value} from './types';
+
+type KeyValueMap = {
+    [TKey in Key | CollectionKey]: Value[TKey];
+};
```

Comment:
is `KeyValueMap` different from `Value`?

Reply:
@hayata-suenaga Yes, this type is used when I need an object which can have zero or more Onyx keys with its associated values, it's used in `multiSet` and `init`.

Reply:
understand but why `multiSet` and `init` cannot use `Value`?

Reply:
I still don't see the difference between `KeyValueMap` and `Value`

Reply:
@hayata-suenaga @blazejkustra `KeyValueMap` is necessary because it ensures type matching between the Onyx key and its value, if using `Value` we will have problems like this:

```ts
    Onyx.multiSet({
        [ONYXKEYS.NVP_PREFERRED_LOCALE]: 1, // should raise an error because this key requires a string, not a number.
    });
```

Reply:
@hayata-suenaga @blazejkustra Yes it was useless before but with the adjustments I'm making in the code I will use this type instead

```ts
type NullableKeyValueMapping = {
    [TKey in OnyxKey]: KeyValueMapping[TKey] | null;
};
```

It will allow the mapping to accept `null` values in order to be used in `multiSet` and `init`.